op-plasma/daserver.go

```diff
@@ -0,0 +1,185 @@
+package plasma
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-service/rpc"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type KVStore interface {
+	// Get retrieves the given key if it's present in the key-value data store.
+	Get(ctx context.Context, key []byte) ([]byte, error)
+	// Put inserts the given value into the key-value data store.
+	Put(ctx context.Context, key []byte, value []byte) error
+}
+
+type MemStore struct {
+	db   map[string][]byte
+	lock sync.RWMutex
+}
+
+func NewMemStore() *MemStore {
+	return &MemStore{
+		db: make(map[string][]byte),
+	}
+}
+
+// Get retrieves the given key if it's present in the key-value store.
+func (s *MemStore) Get(ctx context.Context, key []byte) ([]byte, error) {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	if entry, ok := s.db[string(key)]; ok {
+		return common.CopyBytes(entry), nil
+	}
+	return nil, ErrNotFound
+}
+
+// Put inserts the given value into the key-value store.
+func (s *MemStore) Put(ctx context.Context, key []byte, value []byte) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	s.db[string(key)] = common.CopyBytes(value)
+	return nil
+}
+
+type DAServer struct {
+	log        log.Logger
+	endpoint   string
+	store      KVStore
+	tls        *rpc.ServerTLSConfig
+	httpServer *http.Server
+	listener   net.Listener
+}
+
+type DAServerOption func(b *DAServer)
+
+func WithLogger(log log.Logger) DAServerOption {
+	return func(b *DAServer) {
+		b.log = log
+	}
+}
+
+func NewDAServer(host string, port int, store KVStore, opts ...DAServerOption) *DAServer {
+	endpoint := net.JoinHostPort(host, strconv.Itoa(port))
+	return &DAServer{
+		log:      log.Root(),
+		endpoint: endpoint,
+		store:    store,
+		httpServer: &http.Server{
+			Addr: endpoint,
+		},
+	}
+}
+
+func (d *DAServer) Start() error {
+	mux := http.NewServeMux()
+
+	mux.Handle("/get/", http.StripPrefix("/get/", http.HandlerFunc(d.HandleGet)))
+	mux.Handle("/put/", http.StripPrefix("/put/", http.HandlerFunc(d.HandlePut)))
```

Comment:
I don't understand the use of `StripPrefix` here, and looking at the mono repo, the only examples of this are in the plasma project. Could you check the other `NewServeMux` projects in the monorepo to see if this is required?